From 68e4bd98e53a45ae6ba91dece73ba866023b57de Mon Sep 17 00:00:00 2001
From: Simon Busch <morphis@gravedo.de>
Date: Tue, 11 Mar 2014 16:09:43 +0100
Subject: [PATCH] Determine devicePixelRatio from environment variable if set

Otherwise it would be the job of the platform plugin to set the right value. Having this
at a central place as option when the platform plugin doesn't override it is what we need.

Signed-off-by: Simon Busch <morphis@gravedo.de>
---
 src/gui/kernel/qplatformwindow.cpp | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/gui/kernel/qplatformwindow.cpp b/src/gui/kernel/qplatformwindow.cpp
index 23e492e..9360821 100644
--- a/src/gui/kernel/qplatformwindow.cpp
+++ b/src/gui/kernel/qplatformwindow.cpp
@@ -398,6 +398,9 @@ void QPlatformWindow::handleContentOrientationChange(Qt::ScreenOrientation orien
 */
 qreal QPlatformWindow::devicePixelRatio() const
 {
+    QByteArray dpr = qgetenv("QT_PLATFORM_DPR");
+    if (dpr.length() > 0)
+        return static_cast<qreal>(dpr.toDouble());
     return 1.0;
 }
 
